### PR TITLE
[IDP-1831] Add LinearB to project

### PR DESCRIPTION
**Pull request by tjosepo**
## Description of changes

Add LinearB to project, to keeping track of deployment time.

~We couldn't use [gsoft-inc/wl-reusable-workflows/.github/workflows/linearb-deployment.yml@main](https://github.com/gsoft-inc/wl-reusable-workflows/blob/main/.github/workflows/linearb-deployment.yml) as-is, because this workflow assumes that the pipeline **always** publishes something. Since we are using Changeset, our pipeline either publishes the packages or creates a PR.~

~Add two composite actions to keep track of the deployment.~

- ~LinearB-Pre simply creates a timestamp for the start of the build.~
- ~LinearB-Main uses that timestamp and sends deployment info to LinearB.~

### CODE REVIEW

File: .github/actions/linearb-main/action.yml
Comment (heqianwang) [inline .github/actions/linearb-main/action.yml:1]: I think you can reuse the linearb reusable workflow defined here: https://github.com/gsoft-inc/wl-reusable-workflows/blob/main/.github/workflows/linearb-deployment.yml

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -29,4 +33,11 @@ jobs:
...
-          NPM_TOKEN: ${{ secrets.WORKLEAP_NPM_TOKEN }}
\ No newline at end of file
+          NPM_TOKEN: ${{ secrets.WORKLEAP_NPM_TOKEN }}
+      
+      - name: LinearB
+        if: ${{ steps.changesets.outputs.published == 'true' }}
+        uses: ./.github/actions/linearb-main
+        with:
+          environment: release
+          timestamp: ${{ steps.linearb-pre.outputs.timestamp }}
```
Comment (heqianwang) [inline .github/workflows/publish.yml:43]: If we use the reusable workflow, you just add it like this:
https://github.com/gsoft-inc/wl-extensions-mongo/blob/fc39caadab7642621ee8b851b8bdb0a9fb8f0601/.github/workflows/publish.yml#L27C3-L32C21

Comment (meziantou) [inline .github/workflows/publish.yml:43]: I don't think you need to bother with the timestamp. Once the PR is merged, you can trigger the linearb steps. The start of the pipeline or this step should almost have the same timestamp. 
I'm not familiar with this flow, so maybe I'm wrong.

So, I think you can reuse the existing shared steps and call it here.

File: .github/actions/linearb-main/action.yml
Comment (tjosepo) [reply .github/actions/linearb-main/action.yml:1]: Does it matter _when_ the timestamp of the build is taken?

Does it have to be at the start of the build (to measure duration, etc.) or does it only serve to order the different deployments?

Comment (meziantou) [reply .github/actions/linearb-main/action.yml:1]: The timestamp should be when the project is deployed. So, most of the time, `now`.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (meziantou) [reply .github/workflows/publish.yml:43]: If the timestamp is important (see other comments), I would prefer to edit the [shared step](https://github.com/gsoft-inc/wl-reusable-workflows/blob/main/.github/workflows/linearb-deployment.yml) to add an optional parameter. So, you could avoid duplication.

Comment (tjosepo) [reply .github/workflows/publish.yml]: No, I'm pretty sure I'm the one who did the wrong thing by assuming the timestamp had to be at the beginning of the workflow. I should be able to reuse the shared workflow.

**Comment by tjosepo** [comment]
Updated to use the shared workflow. The job only runs conditionally if there is actually a publish to npm.

**Review by heqianwang — APPROVED** [review]

**Pull request merged**